### PR TITLE
Burger menu

### PR DESCRIPTION
**Pull request by StarijShlupick**
just tried to use dynamic components and angular animations

### CODE REVIEW

File: rsclone/src/app/components/burger-menu/burger-menu.component.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        style({
+          opacity: '0'
+        }),
+        animate('250ms ease-out')
+      ])
+    ])
+  ]
+})
+export class BurgerMenuComponent {
+  @Output() close = new EventEmitter<void>();
```
Comment (annaStoma) [inline rsclone/src/app/components/burger-menu/burger-menu.component.ts]: тип для `close`

File: rsclone/src/app/components/menu-button/menu-button.component.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<ng-template appBurgerMenu></ng-template>
```
Comment (annaStoma) [inline rsclone/src/app/components/menu-button/menu-button.component.html:1]: `<ng-template #appBurgerMenu></ng-template>`

File: rsclone/src/app/components/menu-button/menu-button.component.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  templateUrl: './menu-button.component.html',
+  styleUrls: ['./menu-button.component.scss']
+})
+export class MenuButtonComponent implements OnInit {
+
+  @ViewChild(BurgerDirective) burgerDir: BurgerDirective;
+
+  constructor(private resolver: ComponentFactoryResolver) { }
+
+  ngOnInit(): void {
```
Comment (annaStoma) [inline rsclone/src/app/components/menu-button/menu-button.component.ts]: `ngOnInit` можно убрать

File: rsclone/src/app/models/burger.directive.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import {Directive, ViewContainerRef} from '@angular/core';
+
+@Directive({
+  selector: '[appBurgerMenu]'
+})
+export class BurgerDirective {
+  constructor(public containerBurgerMenu: ViewContainerRef) {
```
Comment (annaStoma) [inline rsclone/src/app/models/burger.directive.ts:7]: для чего эта директива?

**Review by maxim-tihomirov — APPROVED** [review]

File: rsclone/src/app/components/menu-button/menu-button.component.ts (hunk 3 above)
Comment (StarijShlupick) [reply rsclone/src/app/components/menu-button/menu-button.component.ts]: fixed

File: rsclone/src/app/components/burger-menu/burger-menu.component.ts (hunk 1 above)
Comment (StarijShlupick) [reply rsclone/src/app/components/burger-menu/burger-menu.component.ts]: fixed: @Output() close: EventEmitter`<void>` = new EventEmitter`<void>`();

File: rsclone/src/app/models/burger.directive.ts (hunk 4 above)
Comment (StarijShlupick) [reply rsclone/src/app/models/burger.directive.ts:7]: This directive shows us where is containerBurgerMenu placed. So its allows us to specify a place in the template where burger-menu component should be rendered.

Comment (StarijShlupick) [reply rsclone/src/app/models/burger.directive.ts:7]: > для чего эта директива?

This directive shows us where is containerBurgerMenu placed. So its allows us to specify a place in the template where burger-menu component should be rendered.

**Review by Polkhoovskaya — APPROVED** [review]

**Pull request merged**